tests/upgrades/test_errata.py

```diff
@@ -322,3 +326,270 @@ def test_post_scenario_errata_count_installtion(self):
         )
         for package in FAKE_9_YUM_UPDATED_PACKAGES:
             self._check_package_installed(client_container_id, package)
+
+
+class Scenario_errata_count_with_previous_version_katello_agent(APITestCase):
+    """The test class contains pre and post upgrade scenarios to test erratas count
+    and remotely install using n-1 'katello-agent' on content host.
+
+    Test Steps:
+
+        1. Before Satellite upgrade, Create a content host and register it with
+            Satellite
+        2. Install packages and down-grade them to generate errata.
+        3. Upgrade Satellite
+        4. Check if the Erratas Count in Satellite after the upgrade.
+        5. Install erratas remotely on content host and check the erratas count.
+
+    BZ: 1529682
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.docker_vm = settings.upgrade.docker_vm
+        cls.client_os = DISTRO_RHEL7
+        cls.org = entities.Organization().search(
+            query={'search': 'name="{}"'.format(DEFAULT_ORG)})[0]
+        cls.loc = entities.Location().search(
+            query={'search': 'name="{}"'.format(DEFAULT_LOC)})[0]
+
+    def _run_goferd(self, client_container_id):
+        """Start the goferd process."""
+        kwargs = {'async': True, 'host': self.docker_vm}
+        execute(
+            docker_execute_command,
+            client_container_id,
+            'pkill -f gofer',
+            **kwargs
+        )
+        execute(
+            docker_execute_command,
+            client_container_id,
+            '/usr/bin/goferd -f',
+            **kwargs
+        )
+        status = execute(docker_execute_command, client_container_id, 'ps -ef',
+                         host=self.docker_vm)[self.docker_vm]
+        self.assertIn('goferd', status)
+
+    def _check_package_installed(self, client_container_id, package):
```

Comment:
@vijay8451 Can we create a common class for storing all repetitive methods in this file and then using them both required class. Earlier it was only a single class using the same methods hence it was not in common class, but now we have both classes using almost the same methods. what do you think?

Reply:
@omkarkhatavkar Yes I agree with it and noticed the same ... however this work require some refactor with exists tests as well hence I am thinking handle under separate issue/PR #https://github.com/SatelliteQE/robottelo/issues/7230  for them

Reply:
OK